digest/digest.go

```diff
@@ -123,6 +123,20 @@ func (d Digest) Algorithm() Algorithm {
 	return Algorithm(d[:d.sepIndex()])
 }
 
+// Verifier returns a writer object that can be used to verify a stream of
+// content against the digest. If the digest is invalid, an error will be
+// returned.
+func (d Digest) Verifier() (Verifier, error) {
+	if err := d.Validate(); err != nil {
```

Comment:
The only downside is that invalid digest will panic, but you should have validated, so let's not create cost if there isn't a sec issue.